models/resnet/resnet.hpp

```diff
@@ -0,0 +1,485 @@
+                           const size_t outSize,
+                           const size_t strideWidth = 1,
+                           const size_t strideHeight = 1,
+                           const size_t kernelWidth = 3,
+                           const size_t kernelHeight = 3,
+                           const size_t padW = 1,
+                           const size_t padH = 1,
+                           const bool downSample = false,
+                           const size_t downSampleInputWidth = 0,
+                           const size_t downSampleInputHeight = 0)
+  {
+    size_t convInputWidth = inputWidth;
+    size_t convInputHeight = inputHeight;
+
+    if (downSample)
+    {
+      convInputWidth = downSampleInputWidth;
+      convInputHeight = downSampleInputHeight;
+    }
+
+    baseLayer->Add(new ann::Convolution<>(inSize, outSize, kernelWidth,
+        kernelHeight, strideWidth, strideHeight, padW, padH, convInputWidth,
+        convInputHeight));
+    mlpack::Log::Info << "Convolution: " << "(" << inSize << ", " <<
+        convInputWidth << ", " << convInputHeight << ")" << " ---> (";
+
+    if (!downSample)
+    {
+      // Updating input dimensions.
+      inputWidth = ConvOutSize(convInputWidth, kernelWidth, strideWidth, padW);
+      inputHeight = ConvOutSize(convInputHeight, kernelHeight, strideHeight,
+          padH);
+    }
+
+    mlpack::Log::Info << outSize << ", " << convInputWidth << ", " <<
+        convInputHeight << ")" << std::endl;
+
+    baseLayer->Add(new ann::BatchNorm<>(outSize, 1e-5));
+    mlpack::Log::Info << "BatchNorm: " << "(" << outSize << ")" << " ---> ("
+        << outSize << ")" << std::endl;
+  }
+
+  /**
+   * Adds downSample block.
+   *
+   * @tparam AddMergeType Layer type in which downSample block will
+   *     be added.
+   *
+   * @param inSize Number of input maps.
+   * @param outSize Number of output maps.
+   * @param downSampleInputWidth Input widht for down-sample block.
+   * @param downSampleInputHeight Input height for down-sample block.
+   * @param kernelWidth Width of the filter/kernel.
+   * @param kernelHeight Height of the filter/kernel.
+   * @param strideWidth Stride of filter application in the x direction.
+   * @param strideHeight Stride of filter application in the y direction.
+   * @param padW Padding width of the input.
+   * @param padH Padding height of the input.
+   */
+  template <typename AddMergeType = ann::AddMerge<>>
+  void DownSample(AddMergeType* resBlock,
+                  const size_t inSize,
+                  const size_t outSize,
+                  const size_t downSampleInputWidth,
+                  const size_t downSampleInputHeight,
+                  const size_t kernelWidth = 1,
+                  const size_t kernelHeight = 1,
+                  const size_t strideWidth = 2,
+                  const size_t strideHeight = 2,
+                  const size_t padW = 0,
+                  const size_t padH = 0)
```

Comment:
These params aren't being used for Downsample block i.e. kernelWidth and Height are always 1 (I guess) and padding is always zero.